zerver/webhooks/bitbucket3/view.py

```diff
@@ -0,0 +1,161 @@
+    commit_url += "commits/%s" % (sha,)
+    body = get_commits_comment_action_message(user_name=user_name,
+                                              action=action,
+                                              commit_url=commit_url,
+                                              sha=sha,
+                                              message=payload["comment"]["text"])
+    return [{"subject": subject, "body": body}]
+
+def repo_forked_handler(payload: Dict[str, Any]) -> List[Dict[str, str]]:
+    repo_name = payload["repository"]["origin"]["name"]
+    subject = BITBUCKET_TOPIC_TEMPLATE.format(repository_name=repo_name)
+    body = BITBUCKET_FORK_BODY.format(
+        display_name=payload["actor"]["displayName"],
+        username=payload["actor"]["name"],
+        fork_name=payload["repository"]["name"],
+        fork_url=payload["repository"]["links"]["self"][0]["href"]
+    )
+    return [{"subject": subject, "body": body}]
+
+def repo_modified_handler(payload: Dict[str, Any]) -> List[Dict[str, str]]:
+    subject_new = BITBUCKET_TOPIC_TEMPLATE.format(repository_name=payload["new"]["name"])
+    body = BITBUCKET_REPO_UPDATED_CHANGED.format(
+        actor=payload["actor"]["name"],
+        change="name",
+        repo_name=payload["old"]["name"],
+        old=payload["old"]["name"],
+        new=payload["new"]["name"]
+    )  # As of writing this, the only change we'd be notified about is a name change.
+    return [{"subject": subject_new, "body": body}]
+
+def repo_push_branch_data(payload: Dict[str, Any], change: Dict[str, Any]) -> Dict[str, str]:
+    event_type = change["type"]
+    repo_name = payload["repository"]["name"]
+    user_name = payload["actor"]["name"]
+    branch_name = change["ref"]["displayId"]
+    branch_head = change["toHash"]
+
+    if event_type == "ADD":
+        body = get_create_branch_event_message(user_name=user_name, url=None, branch_name=branch_name)
+    elif event_type == "UPDATE":
+        body = BRANCH_UPDATED_MESSAGE_TEMPLATE.format(user_name=user_name,
+                                                      branch_name=branch_name,
+                                                      head=branch_head)
+    elif event_type == "DELETE":
+        body = get_remove_branch_event_message(user_name, branch_name)
+    else:
+        raise UnexpectedWebhookEventType("BitBucket3", event_type)
+
+    subject = TOPIC_WITH_BRANCH_TEMPLATE.format(repo=repo_name, branch=branch_name)
+    return {"subject": subject, "body": body}
+
+def repo_push_tag_data(payload: Dict[str, Any], change: Dict[str, Any]) -> Dict[str, str]:
+    event_type = change["type"]
+    repo_name = payload["repository"]["name"]
+    tag_name = change["ref"]["displayId"]
+    user_name = payload["actor"]["name"]
+
+    if event_type == "ADD":
+        action = "pushed"
+    elif event_type == "DELETE":
+        action = "removed"
+    else:
+        raise UnexpectedWebhookEventType("BitBucket3", event_type)
```

Comment:
Also, you should pass in `BitBucket Server` as the first argument to this exception, not `BitBucket3`.

Reply:
My comments above apply here too.

Reply:
Fixed.